eth/api.go

```diff
@@ -694,6 +695,12 @@ func (s *PublicBlockChainAPI) doCall(args CallArgs, blockNr rpc.BlockNumber) (st
 	}
 	stateDb = stateDb.Copy()
 
+	// If there's no code to interact with, respond with an appropriate error
+	if args.To != nil {
+		if code := stateDb.GetCode(*args.To); len(code) == 0 {
+			return "0x", nil, bind.ErrNonExistentContract
```

Comment:
I don't like the cross-dependency on `accounts/abi/bind`. Maybe the error can live in package eth instead.